File: Real-Time-Search-Workshop/javascript/all_labs.js
/**
 * @fileoverview Executes all labs
 * @maker Joey Whelan
 */

import { Lab1 } from "./lab1/lab1.js";
import { Lab2 } from "./lab2/lab2.js";
import { Lab3 } from "./lab3/lab3.js";
import { Lab4 } from "./lab4/lab4.js";
import { Lab5 } from "./lab5/lab5.js";

(async () => {
    const lab1 = new Lab1();
    const client = await lab1.run();

    const lab2 = new Lab2();
    await lab2.run(client);

    const lab3 = new Lab3();
    await lab3.run(client);

    const lab4 = new Lab4();
    await lab4.run(client);

    const lab5 = new Lab5();
    await lab5.run(client);
    
    await client.disconnect();
})();

File: Real-Time-Query-Workshop/java/code/src/main/java/com/redis/queryworkshop/AllLabs.java
package com.redis.queryworkshop;
import redis.clients.jedis.JedisPooled;

public class AllLabs {
    public static void main(String[] args) {
        Lab1 lab1 = new Lab1();
        JedisPooled client = lab1.run();

        Lab2 lab2 = new Lab2();
        lab2.run(client);

        Lab3 lab3 = new Lab3();
        lab3.run(client);

        Lab4 lab4 = new Lab4();
        lab4.run(client);

        Lab5 lab5 = new Lab5();
        lab5.run(client);

        Lab6 lab6 = new Lab6();
        lab6.run(client);
    }
    
}
